tests/integration-tests/tests/scaling/test_scaling.py

```diff
@@ -113,112 +121,291 @@ def test_nodewatcher_terminates_failing_node(scheduler, region, pcluster_config_
-    # Run job on all nodes
-    job_id = _submit_sleep_job(scheduler_commands, num_compute_nodes)
-    # Set n-1 nodes to drain
-    nodes_to_remove = compute_nodes[:-1]
-    nodes_to_retain = compute_nodes[-1:]
-    _set_nodes_to_suspend_state_manually(scheduler_commands, nodes_to_remove)
-    # assert all nodes are retained correctly
-    _assert_nodes_not_terminated_by_nodewatcher(scheduler_commands, compute_nodes)
-    # wait until the job is completed and check that the drain nodes are then terminated
+    job_id = submit_initial_job(
+        scheduler_commands,
+        "sleep 500",
+        partition,
+        dynamic_instance_type,
+        num_dynamic_nodes,
+        other_options="--no-requeue",
+    )
+    static_nodes, dynamic_nodes = assert_initial_conditions(
+        scheduler_commands, num_static_nodes, num_dynamic_nodes, partition
+    )
+    # Set all nodes to drain, static should be in DRAINED and dynamic in DRAINING
+    _set_nodes_to_suspend_state_manually(scheduler_commands, static_nodes + dynamic_nodes)
+    # Static nodes in DRAINED are immediately replaced
+    _wait_for_node_reset(scheduler_commands, static_nodes=static_nodes, dynamic_nodes=[])
+    # Assert dynamic nodes in DRAINING are not terminated during job run
+    _assert_nodes_not_terminated(scheduler_commands, dynamic_nodes)
+    # wait until the job is completed and check that the DRAINING dynamic nodes are then terminated
     scheduler_commands.wait_job_completed(job_id)
     scheduler_commands.assert_job_succeeded(job_id)
-    _assert_nodes_removed_and_replaced_in_scheduler(
-        scheduler_commands, nodes_to_remove, nodes_to_retain, desired_capacity=num_compute_nodes
+    _wait_for_node_reset(scheduler_commands, static_nodes=[], dynamic_nodes=dynamic_nodes)
+    assert_num_instances_in_cluster(cluster_name, region, len(static_nodes))
+
+
+def _test_computemgtd_logic(
+    remote_command_executor,
+    scheduler_commands,
+    cluster_name,
+    region,
+    test_datadir,
+    partition,
+    num_static_nodes,
+    num_dynamic_nodes,
+    dynamic_instance_type,
+):
+    """Test that computemgtd is able to shut nodes down when clustermgtd and slurmctld are offline."""
+    logging.info("Testing that nodes are shut down when clustermgtd and slurmctld are offline")
+    submit_initial_job(
+        scheduler_commands,
+        "sleep 500",
+        partition,
+        dynamic_instance_type,
+        num_dynamic_nodes,
+        other_options="--no-requeue",
+    )
+    assert_initial_conditions(scheduler_commands, num_static_nodes, num_dynamic_nodes, partition)
+    logging.info("Killing clustermgtd and slurmctld")
+    remote_command_executor.run_remote_script(
+        str(test_datadir / "slurm_kill_clustermgtd_slurmctld.sh"), run_as_root=True
     )
+    logging.info("Waiting for computemgtd to self-terminate all instances")
+    wait_for_num_instances_in_cluster(cluster_name, region, 0)
```

Comment:
if we want to speed this up especially now that we increase the timeout to 5 mins we could manually change the timestamp in the clustermgtd hearthbeat file. As an additional enhancement we could check the node is kept with only clustermgtd down but is terminated when also slurm is down

Reply:
Done